src/utils/getFilteredPeople.ts

```diff
@@ -0,0 +1,48 @@
+import { Person } from '../types';
+
+export const getFilteredPeople = (
+  people: Person[],
+  query: string | null,
+  sex: string | null,
+  centuries: string[],
+) => {
+  let filteredPeople = [...people];
+
+  if (query) {
+    const regex = new RegExp(query, 'i');
+
+    filteredPeople = filteredPeople
+      .filter(({ name, fatherName, motherName }) => (
+        regex.test(
+          `${name} ${fatherName || ''} ${motherName || ''}`,
+        )
+      ));
+  }
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter((person) => {
+      switch (sex) {
+        case 'm':
+          return person.sex === 'm';
+        case 'f':
+          return person.sex === 'f';
+        default:
+          return true;
+      }
+    });
+  }
+
+  if (centuries.length) {
```

Comment:
I guess, in this case we should take just born cuntury into account, you can see the difference between that filter and filter on demo link

Reply:
you are right, thanks